provider/core/src/serde/de.rs

```diff
@@ -46,62 +25,55 @@ where
         DeserializingBufferProvider(self)
     }
 }
-
-fn deserialize_impl<'data, M>(
```

Comment:
Issue: Please restore this as a regular function. I wrote it as a regular function because

1. It is a complex function and I strongly prefer for complex functions to be their own function
2. It reduces indentation which allows for more content on each line
3. This is the type of function that may often show up in stack traces and it's helpful if it has its own name
4. The fact that the function has a type argument is more clear
5. If we needed to share this function, we could do so as a standalone function
6. During DCE, I'd be happy if all of the DataProvider code got inlined, but I think it may be useful to keep this as the "terminal" function that doesn't get inlined, especially if we need to deserialize the same type `M` in multiple places in the same program